agent/agent.go

```diff
@@ -237,6 +237,57 @@ func (a *Agent) GetRunningOutputPlugins() []string {
 	return res
 }
 
+type MapFieldSchema struct {
```

Comment:
Quick question: Not super against these two structs, but why are they necessary? Like they're essentially just wrapper structs, but they don't really add any functionality.

Reply:
I think we just added them for clarity, so that when the schema is sent out it looks like {"Key":"foo", "Value":"bar"}

Reply:
For example, "Types" for the influxdb_v2 output plugin is:
{"Bucket":"string",
"BucketTag":"string",
"ClientConfig":
>{"InsecureSkipVerify":"bool","SSLCA":"string","SSLCert":"string","SSLKey":"string","TLSCA":"string","TLSCert":"string","TLSKey":"string"},"ContentEncoding":"string","ExcludeBucketTag":"bool","HTTPHeaders":**{"Value":"string","Key":"string"}**"HTTPProxy":"string","Organization":"string","Timeout":{"Duration":"int64"},"Token":"string","URLs":{"Value":"string","Length":0},"UintSupport":"bool","UserAgent":"string"},

Reply:
Yeah but can't you just have a ```map[string]string``` of the keys and values? 

Why is it beneficial to have ```HTTPHeaders``` be a list of maps that are ```{ "Key": xxx, "Value": yyy }``` instead of just having it be ``` { "Header1": val1, "Header2": val2, etc... } ```

Reply:
First to clarify, `HTTPHeaders` is a map (not a list) and each key-value pair is expecting `{string:string}`. For this example we don't know what headers the user may want included -- it's more of an open-ended value that is to be provided so that the user can just add custom k/v pairs as long as they fit according to the types provided. I imagine in the web interface there's an "Add Field" button you can click to add new k/v pairs, and since the value has type string (as opposed to potentially an array or another map) the web interface would know to display two text fields for each pair added.

I agree that the format is pretty confusing though, and also I can't imagine a case where the key would not be a string. Perhaps to improve we should somehow return

Reply:
Also, are wrapper structs bad? The reasons why I personally thought it was a good idea is that

 - Helps with typing in the code editor
 - Acts as psuedo-documentation

Reply:
Oh so HTTPHeaders uses this format I see. Okay then this should be okay. I thought you guys were designing it this way